mteb/evaluation/evaluators/ClassificationEvaluator.py

```diff
@@ -272,32 +271,31 @@ def __call__(
             verbose=1 if logger.isEnabledFor(logging.DEBUG) else 0,
         )
         X_train = model.encode(
-            self.sentences_train,
+            DataLoader(self.train_dataset),
             task_name=self.task_name,
             **encode_kwargs,
         )
         if test_cache is None:
-            X_test = model.encode(
-                self.sentences_test,
+            test_cache = model.encode(
+                DataLoader(self.eval_dataset),
```

Comment:
```suggestion
                DataLoader(self.eval_dataset.with_format("torch"), **dataloader_kwargs),
```

Not sure we need `with_format("torch")`

Reply:
I think so, because with it, the dataloader will output `Tensors` instead of lists. This will be helpful for image tasks.
https://huggingface.co/docs/datasets/en/use_with_pytorch#dataset-format

Reply:
but it does not seem to be required for text:

```py
import torch
ds = load_dataset('DDSC/lcc', split='train', streaming=True) # small ds
ds["text"]
dataloader = torch.utils.data.DataLoader(ds.with_format("torch"), num_workers=0)
sample = next(iter(dataloader))
sample # <-- same regardless of with_format
```

Reply:
For texts yes, but image would require this
```python
from datasets import load_dataset
import torch
dataloader = torch.utils.data.DataLoader(ds.with_format("torch"))
dataset = load_dataset("ylecun/mnist")["test"]
dataloader = torch.utils.data.DataLoader(ds.with_format("torch"), num_workers=0)
sample = next(iter(dataloader))
# {'image': tensor([[[[  0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,
#             0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0],
```

Reply:
Agree, but for text tasks we can remove it (but for anything general it we can keep it there)